src/dev/getting_started.md

```diff
@@ -0,0 +1,100 @@
+# Building and Running Tests
+
+## Repos
+
+Yorick is spread over a handful of repos, but the two you are likely to need to clone are:
+
+ - [ykrustc](https://github.com/softdevteam/ykrustc): The compiler.
+ - [yk](https://github.com/softdevteam/ykrustc): The runtime parts of the system.
+
+The latter is a monorepo containing a few different crates.
+
+## Building and Testing the Compiler
+
+### Building the Compiler
+
+To build the compiler, run:
+```
+./x.py build --stage 1
+```
+
+A stage one build is usually sufficient for development.
+
+If you have made changes to `ykpack` (in the `yk` repo), then edit the
+`Cargo.toml` in the top level of `ykrustc` and uncomment (and adapt if needed)
+the lines:
+
+```
+#[patch."https://github.com/softdevteam/yk"]
+#ykpack = { path = "../yk/ykpack" }
+```
+
+This will override the compiler's dependency to your local version.
+
+### Testing the Compiler
+
+The compiler is tested with:
+```
+./x.py test --stage 1
+```
+
+You can do more thorough testing by testing stage 2, but it takes quite a
+while. CI will always to a full stage 2 test.
+
+### Build Configuration
+
+You can configure the compiler by placing a config.toml in the top-level of the
+`ykrustc` directory. `config.toml.example` is upstream's example config.
+`.buildbot.config.toml` is the one CI uses.
+
+## Building and Testing in `yk`
+
+To work on `yk` you will need to have built the compiler, as detailed above.
+Then the easiest way to get going is to use `rustup` to create a "linked
+toolchain" and then override the `yk` repo to use it.
+
+Supposing you have a stage 1 compiler built, you can make a linked toolchain with:
+```
+rustup toolchain link yk-stage1 /path/to/ykrustc/build/x86_64-unknown-linux-gnu/stage1
+```
+
+Then change directory into where you have cloned `yk` and run:
+```
+rustup override set yk-stage1
+```
+
+Now `cargo` will run our compiler (also setting all of the various flags
+required) for this directory instead of the default Rust compiler.
+
+Then you can use normal `cargo` commands, but make sure to set
+`RUSTFLAGS='-C tracer=hw'` to build for hardware tracing.
+
+### Gotchas / Tips
+
+ - If you change the ABI of any of the structures in `yk` then you will have to
+   re-build the compiler before working on `yk` again. Failure to do this leads
+   to weird deserialisation errors.
+
+ - `cargo` will not trigger a rebuild of `yk` if the compiler is rebuilt, so
+   you may have to `cargo clean`.
+
+ - If you want to run `rustfmt`, it's usually easier to invoke the one from the
```

Comment:
Run `rustfmt` on which repo?

Reply:
Will clarify.

Reply:
Fixed in 8d65940ecc47bbac5cbc0fc8f75f2e219a6f5850